src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two',
+      );
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Circle implements Figure {}
+  color: 'red' | 'green' | 'blue';
```

Comment:
Use ur `union` type in all such cases!
```suggestion
  color: Color;
```